### ISSUE TITLE
Logback configuration error detected when using openFeign

### ISSUE THREAD

**Issue by JoniSykes** [OpenFeign/feign#1052]
Trying to use openFeign as our rest client. 
Followed the various guides online to set up. Tried using slf4j plugin too.
No matter what I try AppContext won't load and I see the following error:

```Caused by: java.lang.IllegalStateException: Logback configuration error detected:

ERROR in ch.qos.logback.core.joran.action.AppenderRefAction - Could not find an appender named [loggingAppender_IS_UNDEFINED]. Did you define it below instead of above in the configuration file?
```

application.yaml:

```
logging.level.com.moneysupermarket.energyrequoteconsumer.token.TokenServiceClient: DEBUG

feign:
  client:
    config:
      TokenServiceClient:
        connectTimeout: 5000
        readTimeout: 5000
        loggerLevel: basic
```

service:
```
    public TokenService(...) {
        ...
        this.client = Feign.builder()
                .decoder(new GsonDecoder())
                .logger(new Slf4jLogger(TokenServiceClient.class))
                .target(TokenServiceClient.class, baseURL);
    }

    public String getAutoLoginToken(...) {
        ...
        return client.getToken(request);
    }
```

client:
```
@FeignClient(value = "tokenServiceClient")
public interface TokenServiceClient {
    @RequestMapping(method = RequestMethod.POST, value = "/tokens")
    @Headers("Content-Type: application/json")
    String getToken(AutoLoginTokenRequest request);
}
```

build.gradle
```
...
"org.springframework.cloud:spring-cloud-starter-openfeign:2.1.2.RELEASE",
"org.springframework.cloud:spring-cloud-dependencies:Greenwich.SR2",
"com.netflix.feign:feign-gson:7.2.1",
"io.github.openfeign:feign-slf4j:10.3.0"
```

**Comment by JoniSykes** [comment]
Thanks @kdavisk6. 
That's where I started looking into it too. Isn't it odd though that I don't see any such error when I take out OpenFeign? And by that I mean I'm not changing any logback config at any point. All I'm doing is removing the above code and the build.gradle dependencies. 

I'll add the build.gradle lines above too.

**Comment by kdavisk6** [comment]
@JoniSykes Were you able to resolve this issue?

**Comment by masterhead** [comment]
Hi everyone, for those who still experience this and similar problems with logback, I wanted to share my solution. I had experienced a similar problem with the springProperty feature in logback configuration and after my investigations I found out that it wasn't because of feign, it was because of spring-cloud-starter dependency added with spring-cloud-starter-openfeign dependency. We didn't have spring-cloud dependency before, until we've added spring-cloud-starter-openfeign dependency to the project. The problem occurs because of a change in the way that spring-cloud initializes the logback configuration: With only spring-boot, it was able to load properties in application.yml files, but after adding the spring-cloud dependency, it was not able to load these properties from these files. However, it was able to load the properties from bootstrap.yml file. My solution to fix this problem with the properties from application.yml, is to initialize logging after the spring properties are loaded in the context, using `logging.config` in the spring property file, as shown here: https://stackoverflow.com/questions/33505624/logback-and-spring-boots-new-springproperty-lookup-mechanism-not-working/48383482#48383482
